### PR TITLE
system-variables: add docs for new variable `tidb_opt_objective`

### PR DESCRIPTION
**Pull request by time-and-fate**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/43988

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -3146,6 +3146,16 @@ mysql> desc select count(distinct a) from test.t;
...
 
+### `tidb_opt_objective` <span class="version-mark">从 v7.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值：`moderate`
+- 可选值：`moderate`、`determined`
+- 表示优化器在生成执行计划时应该更保守、稳定还是利用更多信息尝试生成更优的计划。
+- 目前该变量只控制优化器是否利用实时统计信息来生成执行计划。实时统计信息是 TiDB 根据 DML 语句自动更新的表的总行数以及修改的行数。默认情况下，TiDB 会基于实时统计信息来生成执行计划。该变量设为 `determined` 后，TiDB 将不再使用这些信息。
```
Comment (songrijie) [inline system-variables.md:3157]:
```suggestion
- 设置优化器优化目标。 `moderate` 维持旧版本的默认行为，优化器会利用更多信息尝试生成更优的计划；`determined` 则倾向于保守，保持执行计划稳定。
- 实时统计信息是 TiDB 在运行时根据 DML 语句自动更新的表的总行数以及修改的行数。保持默认 `moderate` 的设置，TiDB 会基于实时统计信息来调整执行计划。该变量设为 `determined` 后，TiDB 将不再根据运行时信息调整执行计划，执行计划相对稳定。 
- 对于长期稳定的 OLTP 业务，或者如果用户对系统已有的执行计划非常确定，则推荐在 `determined` 模式减少执行计划跳变的可能。 同时还可以结合 [`lock stats`](/sql-statement-lock-stats#lock-stats) ，阻止统计信息的更新，进一步稳定执行计划。 
```

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -3146,6 +3146,17 @@ mysql> desc select count(distinct a) from test.t;
...
 - 表示传输 1 比特数据的网络净开销。该变量是[代价模型](/cost-model.md)内部使用的变量，**不建议**修改该变量的值。
 
+### `tidb_opt_objective` <span class="version-mark">从 v7.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：枚举型
+- 默认值：`moderate`
+- 可选值：`moderate`、`determinate`
+- 该变量用于设置优化器优化目标。`moderate` 维持旧版本的默认行为，优化器会利用更多信息尝试生成更优的计划；`determinate` 则倾向于保守，保持执行计划稳定。
```
Comment (Oreoxmt) [inline system-variables.md:3156]:
```suggestion
- 该变量用于设置优化器优化目标。`moderate` 与 TiDB v7.4.0 之前版本的默认行为保持一致，优化器会利用更多信息尝试生成更优的计划。`determinate` 则倾向于保守，保持执行计划稳定。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by winoros — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -3146,6 +3146,17 @@ mysql> desc select count(distinct a) from test.t;
 - 默认值：`1.0`
 - 表示传输 1 比特数据的网络净开销。该变量是[代价模型](/cost-model.md)内部使用的变量，**不建议**修改该变量的值。
 
+### `tidb_opt_objective` <span class="version-mark">从 v7.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
```
Comment (Oreoxmt) [inline system-variables.md:3152]:
```suggestion
- 是否持久化到集群：是
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
```
@winoros PTAL

**Pull request merged**